hazelcast/src/main/java/com/hazelcast/internal/config/YamlConfigSchemaValidator.java

```diff
@@ -88,4 +96,45 @@ public void validate(YamlMapping rootNode) {
             throw wrap(e);
         }
     }
+
+    private void validateAdditionalProperties(YamlMapping rootNode, String hzConfigRootNodeName) {
+        if (!PERMITTED_ROOT_NODES.contains(hzConfigRootNodeName)) {
+            throw new IllegalArgumentException(hzConfigRootNodeName);
+        }
+        ObjectSchema schema = (ObjectSchema) SCHEMA;
+        Set<String> forbiddenRootPropNames = ((ObjectSchema) schema.getPropertySchemas()
+                .get(hzConfigRootNodeName)).getPropertySchemas().keySet();
+        List<String> misIndentedRootProps = new ArrayList<>();
+        rootNode.children().forEach(yamlNode -> {
+            if (forbiddenRootPropNames.contains(yamlNode.nodeName())) {
+                misIndentedRootProps.add(yamlNode.nodeName());
+            }
+        });
+        if (misIndentedRootProps.isEmpty()) {
+            return;
+        }
+        if (misIndentedRootProps.size() == 1) {
+            String propName = misIndentedRootProps.get(0);
+            throw createExceptionForMisIndentedConfigProp(propName, true);
+        } else {
+            List<SchemaViolationConfigurationException> causes = misIndentedRootProps.stream()
+                    .map(prop -> createExceptionForMisIndentedConfigProp(prop, false))
+                    .collect(toList());
+            throw new SchemaViolationConfigurationException(withNote(causes.size() + " schema violations found"),
+                    "#", "#", causes);
+        }
+    }
+
+    private static String withNote(String originalMessage) {
+        return originalMessage + System.getProperty("line.separator") + "Note: you can disable this validation by passing the "
+                + "-Dhazelcast.yaml.config.indentation.check.enabled=false system property";
```

Comment:
Minor: you can get the property name from the property itself, so if we ever change it for any reason, we won't forget updating this message.

Reply:
Done.